File: app/src/main/java/ru/santaev/clipboardtranslator/api/YandexApi.java
package ru.santaev.clipboardtranslator.api;


import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;
import ru.santaev.clipboardtranslator.BuildConfig;

public interface YandexApi {

    String API_KEY = BuildConfig.YANDEX_TRANSLATE_API_KEY;
    String SERVER_URL = "https://translate.yandex.net/";

    @FormUrlEncoded
    @POST("api/v1.5/tr.json/translate")
    Call<TranslateResponse> translate(@Field("text") String text,
                                      @Field("lang") String lang,
                                      @Field("key") String apiKey);

    @GET("api/v1.5/tr.json/getLangs")
    Call<LanguagesResponse> getLangs(@Query("key") String apiKey,
                                     @Query("ui") String uiLang);
}

File: app/src/main/java/ru/santaev/clipboardtranslator/api/YandexApi.kt
package ru.santaev.clipboardtranslator.api


import io.reactivex.Single
import retrofit2.http.*
import ru.santaev.clipboardtranslator.BuildConfig

interface YandexApi {

    @FormUrlEncoded
    @POST("api/v1.5/tr.json/translate")
    fun translate(@Field("text") text: String,
                  @Field("lang") lang: String,
                  @Field("key") apiKey: String): Single<TranslateResponse>

    @GET("api/v1.5/tr.json/getLangs")
    fun getLangs(@Query("key") apiKey: String,
                 @Query("ui") uiLang: String): Single<LanguagesResponse>

    companion object {

        val API_KEY = BuildConfig.YANDEX_TRANSLATE_API_KEY
        val SERVER_URL = "https://translate.yandex.net/"
    }
}
